### ISSUE TITLE
Lens Metrics - Role Bindings

### ISSUE THREAD

**Issue by NArnott** [lensapp/lens#626]
What role bindings does the user need to have in order to view Lens Metrics?

I have a cluster set up with Lens Metrics and everything works great. I'm now trying to create a limited user group, which only has edit access to the default namespace (and list namespaces). I noticed that when I do this, Lens doesn't display the metrics for those limited users. If I bind the edit role to the lens-metrics namespace, it works, but not the view role. 

So basically, there is something in the system edit role that Lens appears to want access to in order to view Metrics...? What is it? It would be great if Lens output the results of various failed K8s commands so I could figure this out myself, but in the meantime, what am I missing?

**Comment by just1689** [comment]
Howdy

I faced this problem. It ended up being Kubernetes sub-resources. I had to explicitly give it access to quite a few subresource (only with get, list and watch verbs).

Checkout https://gist.github.com/just1689/272f3e78a540c10b7bda390eef9b4473

You could make a more limited version that you give all your users and then user `Role`s giving particular mutation permissions to a given namespace.

**Comment by Nokel81** [comment]
Closing as answered.

**Comment by NArnott** [comment]
Seems that with Lens 5, what we had set up no longer works. Is there a log or something we can review to see what Lens is trying to access and what it's failing on? 

I'm currently using Lens 5.2.7.

**Comment by Nokel81** [comment]
@NArnott if you run lens from the terminal, does that help?

**Comment by NArnott** [comment]
Not sure I follow. I'm on a windows dev machine running Lens. Lens is installed as a windows app. We don't normally use a CLI to start Lens...

**Comment by Nokel81** [comment]
Yes that is true, but you still can.

**Comment by NArnott** [comment]
Well, just tried, but no difference.

![image](https://user-images.githubusercontent.com/198647/143094523-13081e78-c10f-4948-9e36-e2318d6b50b3.png)

No useful output produced doing it that way.

It works fine when I'm part of the system:masters group. I has issues though once I log in using a restricted user. Is there any documentation as to what Lens needs access to in order for metrics to work so I can provide that to these restricted user groups, while not giving them everything?

An example of a restricted user is one who has full readonly access by creating a ClusterBinding against the ClusterRole "view" to their group. They can view everything, yet can't view metrics.

**Comment by elijah-ward** [comment]
Experiencing this issue as well
